File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let subset=[];
    recur([],0);
    
    function recur(set,step){
        subset.push(set);
        for(let i=step;i<nums.length;i++){
            recur([...set,nums[i]],i+1);
        }
    }
    return subset;
};

File: 0078-subsets/0078-subsets.cs
public class Solution {
    public IList<IList<int>> Subsets(int[] nums) {
        IList<IList<int>> subset= new List<IList<int>>();
        subset.Add(new List<int>());

        for(int i=0;i<nums.Length;i++){
            int count=subset.Count;
            for(int ii=0;ii<count;ii++){
                var hold=subset[ii].ToList();
                hold.Add(nums[i]);
                subset.Add(hold);
            }
        }
        return subset;
    }
}